cmd/fyne/package.go

```diff
@@ -93,42 +93,75 @@ type packager struct {
 }
 
 func (p *packager) packageLinux() error {
-	prefixDir := ensureSubDir(ensureSubDir(p.dir, "usr"), "local")
-	shareDir := ensureSubDir(prefixDir, "share")
+	var prefixDir string
+	local := "local/"
+	tempDir := "tmp-pkg"
 
-	appsDir := ensureSubDir(shareDir, "applications")
-	desktop := filepath.Join(appsDir, p.name+".desktop")
-	deskFile, _ := os.Create(desktop)
-	_, err := io.WriteString(deskFile, "[Desktop Entry]\n"+
-		"Type=Application\n"+
-		"Name="+p.name+"\n"+
-		"Exec="+filepath.Base(p.exe)+"\n"+
-		"Icon="+p.name+"\n")
-	if err != nil {
-		return errors.Wrap(err, "Failed to write desktop entry string")
+	if p.install {
+		tempDir = ""
+	}
+
+	if _, err := os.Stat(filepath.Join("usr", "local")); os.IsNotExist(err) {
+		prefixDir = ensureSubDir(ensureSubDir(p.dir, tempDir), "usr")
+		local = ""
+	} else {
+		prefixDir = ensureSubDir(ensureSubDir(ensureSubDir(p.dir, tempDir), "usr"), "local")
 	}
 
+	shareDir := ensureSubDir(prefixDir, "share")
+
 	binDir := ensureSubDir(prefixDir, "bin")
 	binName := filepath.Join(binDir, filepath.Base(p.exe))
-	err = copyExeFile(p.exe, binName)
+	err := copyExeFile(p.exe, binName)
 	if err != nil {
-		return errors.Wrap(err, "Failed to copy exe file")
+		return errors.Wrap(err, "Failed to copy application binary file")
 	}
 
-	iconThemeDir := ensureSubDir(ensureSubDir(shareDir, "icons"), "hicolor")
-	iconDir := ensureSubDir(ensureSubDir(iconThemeDir, "512x512"), "apps")
+	iconDir := ensureSubDir(shareDir, "pixmaps")
 	iconPath := filepath.Join(iconDir, p.name+filepath.Ext(p.icon))
 	err = copyFile(p.icon, iconPath)
 	if err != nil {
 		return errors.Wrap(err, "Failed to copy icon")
 	}
 
+	appsDir := ensureSubDir(shareDir, "applications")
+	desktop := filepath.Join(appsDir, p.name+".desktop")
+	deskFile, _ := os.Create(desktop)
+	_, err = io.WriteString(deskFile, "[Desktop Entry]\n"+
+		"Type=Application\n"+
+		"Name="+p.name+"\n"+
+		"Exec="+filepath.Base(p.exe)+"\n"+
+		"Icon="+p.name+"\n")
+	if err != nil {
+		return errors.Wrap(err, "Failed to write desktop entry string")
+	}
+
 	if !p.install {
-		tarCmd := exec.Command("tar", "zcf", p.name+".tar.gz", "usr")
-		err := tarCmd.Run()
+		makefile, _ := os.Create(filepath.Join(p.dir, tempDir, "Makefile"))
+		_, err := io.WriteString(makefile, `ifneq ("$(wildcard /usr/local)","")`+"\n"+
+			"DESTDIR ?= /usr/local\n"+
+			"else\n"+
+			"DESTDIR ?= /usr\n"+
+			"endif\n"+
+			"\n"+
+			"install:\n"+
+			"	install -Dm00644 usr/"+local+"share/applications/"+p.name+`.desktop $(DESTDIR)$(PREFIX)/share/applications/`+p.name+".desktop\n"+
+			"	install -Dm00644 usr/"+local+"bin/"+filepath.Base(p.exe)+` $(DESTDIR)$(PREFIX)/bin`+filepath.Base(p.exe)+"\n"+
+			"	install -Dm00644 usr/"+local+"share/pixmaps/"+p.name+filepath.Ext(p.icon)+` $(DESTDIR)$(PREFIX)/share/icons/hicolor/512x512/apps/`+p.name+filepath.Ext(p.icon)+"\n")
```

Comment:
This destination directory is not the same as is installed by `fyne install`.
Which is wrong?

Reply:
Yeah looks like I forgot to update the makefile. Good catch :+1: